solr/core/src/java/org/apache/solr/request/json/JsonQueryConverter.java

```diff
@@ -69,6 +69,13 @@ private void buildLocalParams(StringBuilder builder, Object val, boolean isQPars
             "Error when parsing json query, expect only one query parser here, but found : "+map.keySet());
       }
       String qtype = map.keySet().iterator().next();
+      String qTypeTrimmed = qtype.trim();
+      String tagName = null;
+      if (qTypeTrimmed.startsWith("#")) {
+        tagName = qTypeTrimmed.substring(1);
```

Comment:
I suppose it makes more sense to keep hash in tag just to make it more noticeable and explicit.